### DISCUSSION TITLE
Xcode 14.3 이슈 공유

### DISCUSSION THREAD

**Discussion by JIWON1923** [Team-ShortcutsZip/MacC-Team-HappyAnding#435] [General]
## 이슈 상황

Xcode 14.3 이슈 상황 공유드립니다!

엑스코드 업데이트 후 컴파일 에러가 발생합니다.
Optional chaining에서 이슈가 발생하는데, [release note](https://developer.apple.com/documentation/xcode-release-notes/xcode-14_3-release-notes)에서 관련 업데이트 사항은 찾을 수 없어서 공유드립니다. (혹시 찾으신 분은 공유 부탁드려요)

![image](https://user-images.githubusercontent.com/68676844/230566876-2554afed-13e5-4b96-b59b-8eec01f7e712.png)

우선 옵셔널이 아닌데 옵셔널 체이닝이 사용된 부분에서 해당 이슈가 일어나는 것으로 확인했습니다.
우선 bug fix로 해당 이슈 해결한 후 PR 올리겠습니다!

## 에러 메시지
`Initializer for conditional binding must have Optional type, not '[Shortcuts]'

**Comment by HanGyeongjun** [comment]
저도 이번 14.3 업데이트에서 같은 이슈가 발생한 사례나 원인을 찾지 못했습니다 ㅠㅠ 하지만 수정이 필요한 부분은 맞으니 올려주신 PR 빠르게 확인하겠습니다!

**Comment by JIWON1923** [comment]
가장 비슷한 사례로 [Optional chaining array with Xcode 14.3](https://forums.swift.org/t/optional-chaining-array-with-xcode-14-3/64139)가 있네요

**Comment by JIWON1923** [comment]
### 공유
[관련 포럼](https://forums.swift.org/t/optional-chaining-array-with-xcode-14-3/64139/4) 이 부분 확인해 보면 좋을 것 같아요.
예전에 오류를 던지지 않았던 부분에 있어서 개선이 된 것 같네요!
사이트 타고 타서 정보 찾아보는데, 쉽지 않네요... ^^;;


``` swift

import SwiftUI

struct Item : Hashable {
  var question: String
  var answer: Int
}

struct MyView : View {
  var items: [Item] = []
  
  var body: some View {
    ZStack {
      ForEach(items, id: \.self) { item in
        if let question = item.question { 🔴 initializer for conditional binding must have Optional type, not 'String'
          ...
        }
      }
    }
  }
}

```

### Swift 5.8 변경사항
Swift 5.8 버전의 업데이트 중 옵셔널 [관련 이슈](https://github.com/apple/swift-evolution/blob/main/proposals/0375-opening-existential-optional.md) 변경사항입니다.

다운캐스팅 없이 옵셔널을 사용할 수 있도록 변경된 것 같은데, 이런 방식을 채택하기 위해 좀 더 안전한 코드를 유도하도록 수정된걸까요?????
흑흑흑 같이 공부할사람 구함니다
